cosmo/serializer.py

```diff
@@ -87,11 +87,21 @@ def _get_unit(self, iface):
                 )
                 return
             unit_stub["vlan"] = iface["untagged_vlan"]["vid"]
+            l2vpn_vlan_term = self.l2vpn_vlan_terminations.get(iface["untagged_vlan"]["id"])
+        else:
+            l2vpn_vlan_term = False
 
-            l2vpn = self.l2vpn_vlan_terminations.get(iface["untagged_vlan"]["id"]) \
-                    or self.l2vpn_interface_terminations.get(iface["id"])
-            if l2vpn:
+        l2vpn = l2vpn_vlan_term or self.l2vpn_interface_terminations.get(iface["id"])
+        if l2vpn:
+            if l2vpn['type'] == "VPWS":
+                if iface.get("untagged_vlan"):
+                    unit_stub["encapsulation"] = "vlan-ccc"
+                else:
+                    unit_stub["encapsulation"] = "ethernet-ccc"
+            elif l2vpn['type'] in ["MPLS_EVPN", "VXLAN_EVPN"]:
                 unit_stub["encapsulation"] = "vlan-bridge"
+
+            if unit_stub["encapsulation"]:
```

Comment:
What do you want to check here?
Probably something like `if l2vpn['type'] in ['x','y']` or another variable with a describing name would be helpful.

Reply:
Also I think, this variable is set every time for every valid L2VPN we support...